### PR TITLE
Bug 1574648 - add a servo project, plus machinery

### PR DESCRIPTION
**Pull request by djmitche**
* Add servo
* Add a `repo-admin:..` utility role to grant admin access to repos
* Add support for "externally managed" projects, where the project admins are responsible for all or most of the resources in the project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,3 +44,10 @@ Each project has a its own worker pools, and user roles can be given "admin" acc
 Repositories are granted scopes via the [Taskcluster-github scheme](https://docs.taskcluster.net/docs/reference/integrations/github/taskcluster-yml-v1#scopes-and-roles).
 Each repository is associated with a project, and scopes granted to the repository should be associated with that project.
 This occurs within `config/projects.yml`.
+
+### Externally Managed Projects
+
+This respository manages all resources in the deployment *except* those associated with "externally managed" projects.
+Projects that manage their own resources, either by hand or via their own automation, should have the `externallyManaged` attribute set in `config/projects.yml`, otherwise the next run of `tc-admin apply` will delete the project's resoureces!
```
Comment (imbstack) [inline README.md]: resoureces

File: generate/__init__.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+from . import projects, login_identity, grants
+
+
+def manage_resources(resources, externally_managed):
+    """Set up the set of managed resources, specifically excluding resources that
+    are externally managed, but including everything else."""
+    assert externally_managed, "CAUTION: not tested with no externally-managed projects"
+
+    em_bar = '|'.join(externally_managed)
+    resources.manage(r"Role=(?!project:({}):).*".format(em_bar))
```
Comment (SimonSapin) [inline generate/__init__.py]: For Servo I think we’d like `repo:github.com/servo/*` roles to be externally-managed as well.

Comment (SimonSapin) [reply generate/__init__.py]: And `hook-id:project-servo/*` roles too.

Comment (djmitche) [reply generate/__init__.py]: good points, thanks

**Pull request merged**